vpc/service/long_lived_lock_test.go

```diff
@@ -0,0 +1,168 @@
+package service
+
+import (
+	"context"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	vpcapi "github.com/Netflix/titus-executor/vpc/api"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/empty"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var longLivedLockColumns = []string{"id", "lock_name", "held_by", "held_until"}
+
+func generateLockAndRows(t *testing.T, mock sqlmock.Sqlmock) (*vpcapi.Lock, *sqlmock.Rows) {
+	heldUntil := time.Now()
+	protoHeldUntil, err := ptypes.TimestampProto(heldUntil)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	rand.Seed(time.Now().UnixNano())
+	lock := &vpcapi.Lock{
+		Id:        rand.Int63(),
+		LockName:  "branch_eni_associate_nilitem",
+		HeldBy:    "titusvpcservice-cell-instance",
+		HeldUntil: protoHeldUntil,
+	}
+
+	rows := sqlmock.NewRows(longLivedLockColumns).AddRow(
+		lock.GetId(),
+		lock.GetLockName(),
+		lock.GetHeldBy(),
+		heldUntil,
+	)
+
+	return lock, rows
+}
+
+func TestAPIShouldGetLocks(t *testing.T) {
+	db, mock, err := sqlmock.New()
+	if err != nil {
+		t.Fatalf("could not open mock db: %v", err)
+	}
+	defer db.Close()
+
+	expected, rows := generateLockAndRows(t, mock)
+	mock.ExpectQuery("SELECT id, lock_name, held_by, held_until FROM long_lived_locks LIMIT 1000").WillReturnRows(rows)
+
+	service := vpcService{db: db}
+
+	ctx := context.Background()
+	res, err := service.GetLocks(ctx, &empty.Empty{})
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
```

Comment:
One of the tools we use is gotools assert: https://pkg.go.dev/gotest.tools/v3@v3.0.2/assert?tab=doc -- it makes this slightly cleaner, if you want. It also has a package (cmp), that allows you to write things like:

`asssert.Assert(t, cmp.Len(got, 1))`

or

`assert.Assert(t, proto.Equal(expected, got))`

Reply:
Sure. I tend to be a minimalist when it comes to packages but I have no qualm using it so, at the very least, it's consistent .